### ISSUE TITLE
Synced files are wrong when lines are deleted from original files (EOF not updated correctly)

### ISSUE THREAD

**Issue by maxalbert** [datproject/dat#517]
First of all, kudos for all your work on dat! It looks awesome and I'm well excited to give it a try.

I'm running into the following bug. I'm trying to sync two directories `A/` and `B/`, where `A/` contains just one text file `foobar.txt`. Syncing works as expected so long as I only _add_ lines to the file. These changes are reflected correctly in `B/`. However, when I _delete_ lines the synced file in `B/` is wrong - it still contains some lines from the original version at the end that should have been deleted.

Steps to reproduce below.
## Setup

Create the two directories `A/` (the source) and `B/` (the target) to be synced. There is one file called `foobar.txt` in the source directory.

```
$ cat A/foobar.txt
1
2
3
4
5
6
7
8
9
```

Share the source directory:

```
$ cd A
$ dat .
Sharing /path/to/A

Share Link: <dat-link>
The Share Link is secret and only those you share it with will be able to get the files

[=============================>] Added 1 items (18 B/18 B)

Waiting for connections. Watching for updates...
```

In a separate terminal, change into `B/` and download the file.

```
$ cd B
$ dat <dat-link> .
Downloading in /path/to/B

Share Link: <dat-link>
The Share Link is secret and only those you share it with will be able to get the files

[=============================>] Downloaded 1 items (18 B/18 B)

Connected to 1 peers. 

Download Finished. You may exit the process with Ctrl-C.
```

So far, all works as expected. The synced file in `B/` has the correct contents.

```
$ cat B/foobar.txt
1
2
3
4
5
6
7
8
9
```
## Bug

The bug arises when I edit the original file `A/foobar.txt` to delete a bunch of lines.

```
$ cat A/foobar.txt
2
5
8
```

When this new version is saved, dat syncs the directories and the file in `B/foobar.txt` is updated.
The first three lines coincide with the new version (as expected), but the EOF marker doesn't seem to be adjusted accordingly. As a result, the file `B/foobar.txt` still contains all the old lines that came after the third line.

```
$ cat B/foobar.txt
2
5
8
4
5
6
7
8
9
```
## Versions

I'm using dat installed from source (git rev. a7ff2014452397077da0d0016d3807dab9422570).

```
$ dat --version
11.1.2

$ node --version
v4.5.0

$ npm --version
2.15.9

$ nvm --version
0.31.7
```

(Not sure which of the node-related versions are important so I thought I'd paste them all.)

**Comment by joehand** [comment]
Thanks for the awesome bug report! I was able to reproduce and will check it out.

**Comment by karissa** [comment]
@mafintosh do you know why this is happening? Is this related to https://github.com/mafintosh/hyperdrive/pull/99?

**Comment by joehand** [comment]
@maxalbert this should be fixed now, finally! Try out version `12.0.0` and let us know if you still see this issue.
